pkg/querier/worker/scheduler_processor.go

```diff
@@ -123,13 +123,15 @@ func (sp *schedulerProcessor) processQueriesOnSingleStream(workerCtx context.Con
 		}
 
 		if err := sp.querierLoop(execCtx, c, address, inflightQuery); err != nil {
-			// Do not log an error if the query-scheduler is shutting down.
-			if s, ok := status.FromError(err); !ok || !strings.Contains(s.Message(), schedulerpb.ErrSchedulerIsNotRunning.Error()) {
-				level.Error(sp.log).Log("msg", "error processing requests from scheduler", "err", err, "addr", address)
-			}
+			if !grpcutil.IsCanceled(err) {
+				// Do not log an error if the query-scheduler is shutting down.
+				if s, ok := status.FromError(err); !ok || !strings.Contains(s.Message(), schedulerpb.ErrSchedulerIsNotRunning.Error()) {
+					level.Error(sp.log).Log("msg", "error processing requests from scheduler", "err", err, "addr", address)
+				}
 
-			backoff.Wait()
-			continue
+				backoff.Wait()
+				continue
+			}
```

Comment:
I agree with the intent of this change. However, we'll now not log anything if the query is cancelled because the scheduler told the querier to cancel the query, which might make debugging issues harder.

Perhaps we should add an info-level log statement around line 177 below if the query is explicitly cancelled by the scheduler? Even better would be to cancel the context created on line 146 with an error like "query cancelled by scheduler" and then make sure this is logged somewhere or attached to the query trace.

Reply:
Would this happen if the user's client has a timeout shorter than Mimir takes?
This is the case I don't want to log for.
It should be visible as an event in the trace.

Reply:
The querier won't know what caused the cancellation - it could be that the user cancelled the query, or it could be that something in the chain between the user and the querier timed out and cancelled the query, for example.

Reply:
Still sounds like I don't want to see the log here.

Reply:
I think there's value in adding something to the querier's trace span to say "received cancellation from scheduler" - we've had a number of context cancellation-related issues in this part of the codebase lately, and being very explicit like this would have been helpful in debugging the issues.